akka-stream/src/main/scala/akka/stream/javadsl/Graph.scala

```diff
@@ -197,26 +197,36 @@ object Broadcast {
  * '''Completes when''' upstream completes
  *
  * '''Cancels when'''
- *   when one of the downstreams cancel
+ *   when any (eagerCancel=true) or all (eagerCancel=false) of the downstreams cancel
  */
 object Partition {
   /**
-   * Create a new `Partition` stage with the specified input type.
+   * Create a new `Partition` stage with the specified input type, `eagerCancel` is `false`.
    *
    * @param outputCount number of output ports
    * @param partitioner function deciding which output each element will be targeted
    */
   def create[T](outputCount: Int, partitioner: function.Function[T, Integer]): Graph[UniformFanOutShape[T, T], NotUsed] =
-    scaladsl.Partition(outputCount, partitioner = (t: T) ⇒ partitioner.apply(t))
+    new scaladsl.Partition(outputCount, partitioner = (t: T) ⇒ partitioner.apply(t))
```

Comment:
perhaps specify eagerCancel param so we avoid the deprecation warning, or use the apply, as it was

Reply:
I want to avoid having to specify the default value in several places. I avoided the `apply` since using apply to provide the eagerCancel property is not possible, so for symmetry these are all using the ctors.